internal/runbits/dependencies/changesummary.go

```diff
@@ -1,59 +1,28 @@
 package dependencies
 
 import (
-	"encoding/json"
 	"fmt"
 	"sort"
 	"strconv"
 	"strings"
 
 	"github.com/go-openapi/strfmt"
 
-	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/locale"
 	"github.com/ActiveState/cli/internal/logging"
-	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/output"
 	"github.com/ActiveState/cli/internal/sliceutils"
 	"github.com/ActiveState/cli/pkg/buildplan"
+	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/response"
 	"github.com/ActiveState/cli/pkg/platform/model/buildplanner"
 )
 
-type primeable interface {
-	primer.Outputer
-	primer.Auther
-	primer.Projecter
-}
-
 // showUpdatedPackages specifies whether or not to include updated dependencies in the direct
 // dependencies list, and whether or not to include updated dependencies when calculating indirect
 // dependency numbers.
 const showUpdatedPackages = true
 
-func OutputChangeSummary(prime primeable, newCommit *buildplanner.Commit, oldCommit *buildplanner.Commit) error {
-	// Fetch the impact report.
-	beforeExpr, err := json.Marshal(oldCommit.BuildScript())
-	if err != nil {
-		return errs.Wrap(err, "Unable to marshal old buildexpression")
-	}
-
-	afterExpr, err := json.Marshal(newCommit.BuildScript())
-	if err != nil {
-		return errs.Wrap(err, "Unable to marshal buildexpression")
-	}
-	bpm := buildplanner.NewBuildPlannerModel(prime.Auth())
-	params := &buildplanner.ImpactReportParams{
-		Owner:      prime.Project().Owner(),
-		Project:    prime.Project().Name(),
-		BeforeExpr: beforeExpr,
-		AfterExpr:  afterExpr,
-	}
-	report, err := bpm.ImpactReport(params)
-	if err != nil {
-		return errs.Wrap(err, "Failed to fetch impact report")
-	}
-
-	buildPlan := newCommit.BuildPlan()
-
+func OutputChangeSummary(out output.Outputer, report *response.ImpactReportResult, rtCommit *buildplanner.Commit) {
```

Comment:
It looks like we only use `rtCommit` to retrieve the buildplan, so we should just pass the buildplan here.